feat: Add comprehensive Docker support

trose opened the pull request:
# 🐳 Add Docker Support for PotatoMesh

Addresses [GitHub issue #120](https://github.com/l5yth/potato-mesh/issues/120) by providing a complete Docker solution that eliminates complex manual installation.

## 🚀 Key Features

### Multi-Architecture Docker Images
- **Two containers**: Web app (Ruby/Sinatra) and Python ingestor
- **Two architectures**: AMD64, ARM64
- **Published to GHCR**: `ghcr.io/l5yth/potato-mesh/web:latest` and `ghcr.io/l5yth/potato-mesh/ingestor:latest`

### Automated CI/CD
- **GitHub Actions workflows**: Comprehensive testing
- **Security scanning**: Trivy vulnerability scanning
- **Multi-platform builds**: Automated builds and GHCR publishing on version tags

## 📦 Installation

### Quick Start
```bash
# Pull pre-built images
docker pull ghcr.io/l5yth/potato-mesh/web:latest
docker pull ghcr.io/l5yth/potato-mesh/ingestor:latest

# Configure and start
./configure.sh
docker-compose up -d
```

**Transforms PotatoMesh from complex manual installation to simple `docker-compose up -d` experience.**

zmiguel commented on docker-compose.raspberry-pi.yml:
```diff
@@ -0,0 +1,105 @@
+        reservations:
+          memory: 64M
+          cpus: '0.1'
+    healthcheck:
+      test: ["CMD", "pgrep", "-f", "python.*mesh.py"]
+      interval: 60s  # Longer interval for slower hardware
+      timeout: 15s
+      retries: 3
+      start_period: 30s
```
Same here, the image already has a health check configured

zmiguel commented:
after some testing it seems 
`APP_ENV=production` needs to be set on the web container or else we get `attack prevented by Rack::Protection::HostAuthorization` on the webapp when the ingestor tries to send data to it.

trose commented:
@zmiguel Hey I really appreciate the quick feedback. I just incorporated most of your comments. I did just test with my T-deck on MacOS and I may be running into the same docker serial usb access issue. It looks like Docker on MacOS runs inside a linux VM and does not expose serial ports on the host system. I'm going to keep poking at it and see if I can find a workaround or alternative containerization possibly.

For now I just added GH actions that test on windows, linux, and raspberry pi VMs. Hopefully we can get some volunteers to test this on some native hardware ASAP

zmiguel commented:
@trose it should work on linux, I was able to get it working after adding `APP_ENV=production` to the env of web

trose commented on README.md:
```diff
@@ -14,12 +14,118 @@ A simple Meshtastic-powered node dashboard for your local community. _No MQTT cl
 * Allows searching and filtering for nodes in map and table view.
 
-Live demo for Berlin #MediumFast: [potatomesh.net](https://potatomesh.net)
+Live demo: [potatomesh.net](https://potatomesh.net)
```
doh! fix this.

zmiguel commented on .gitignore:
```diff
@@ -62,3 +62,6 @@ coverage/
 reports/
+
+# AI planning and documentation
+ai_docs/
```
Why is this needed?

trose replied:
heh that's where Cursor is hiding it's planning and task docs.

zmiguel commented on docker-compose.dev.yml:
```diff
@@ -0,0 +1,27 @@
+      - ./web:/app
+      - ./data:/data  # Mount data directory for SQL files
+      - /app/vendor/bundle  # Exclude vendor directory from volume mount
+    command: ["ruby", "app.rb", "-p", "41447", "-o", "0.0.0.0"]
```
can be removed since it's defined in the Dockerfile

zmiguel commented on docker-compose.dev.yml:
```diff
@@ -0,0 +1,27 @@
+    volumes:
+      - ./data:/app
+      - /app/.local  # Exclude Python packages from volume mount
+    command: ["python", "mesh.py"]
```
can be removed since it's defined in the Dockerfile

zmiguel commented on docker-compose.yml:
```diff
@@ -0,0 +1,65 @@
+  potatomesh_logs:
+    driver: local
+
+networks:
+  potatomesh-network:
+    driver: bridge
```
this is not needed as all containers defined in the compose file share the same network by default

zmiguel replied:
same applies to the other docker-compose files

zmiguel commented on configure.sh:
```diff
@@ -0,0 +1,135 @@
+read_with_default "Matrix Room (optional, e.g., #meshtastic:matrix.org)" "$MATRIX_ROOM" MATRIX_ROOM
+
+echo ""
+echo "🔐 Security Settings"
```
API key for the map provider can also be asked here

zmiguel reviewed (commented):
Added a few more comments.

Just one extra question:
I see the docker-compose files are all building the images instead of using the image built in the CI, is that on purpose?

trose commented:
Oooookay, I must have had a false-positive with the containerized ingestor on MacOS last night. I've been unable to get the ingestor to recognize the host's usb serial ports after much whittling and poking. I'm updating the readme w/ a note about MacOS & docker.

Good news, I was able to collect live data locally in Denver, CO using the dockerized web app and the ingestor running on native hardware. I collected all of this from T-Deck sitting on my coffee table LOL

<img width="1092" height="529" alt="Screen Shot 2025-09-18 at 10 54 07 AM" src="https://github.com/user-attachments/assets/4b983ddf-4c30-47a8-9985-b4685e7323d4" />

trose commented on README.md:
```diff
@@ -14,12 +14,120 @@ A simple Meshtastic-powered node dashboard for your local community. _No MQTT cl
 * Allows searching and filtering for nodes in map and table view.
 
-Live demo for Berlin #MediumFast: [potatomesh.net](https://potatomesh.net)
+* [ ] Live demo for Berlin #MediumFast: [potatomesh.net](https://potatomesh.net)
```
```suggestion
- Live demo for Berlin #MediumFast: [potatomesh.net](https://potatomesh.net)
```

trose commented:
@zmiguel I updated and simplified the docker-compose ymls to utilize the published images now. I think this should be ready for a final review & testing. I'm looking for my raspberry Pi at home to test the containers on as well...

l5yth reviewed (commented):
Thanks, this is looking good. I left a couple of minor comments.

trose commented:
@l5yth I greatly reduced the README and DOCKER docs down to just necessary details. Restored defaults to project originals/ Berlin too.

You should probably manually trigger the docker GH action to verify that it's working with your account. I ran it with my own ghcr credentials to test then updated it for your's.

trose commented on DOCKER.md:
```diff
@@ -0,0 +1,437 @@
+
+# Copy and configure environment variables
+cp .env.example .env
+# Edit .env with your specific configuration
```
doh! added one

l5yth commented:
Thanks. I'll merge this and will do some tags to test. :)
